go/vt/discovery/topology_watcher.go

```diff
@@ -199,9 +199,6 @@ func (tw *TopologyWatcher) loadTablets() {
 				log.Errorf("cannot get tablet for alias %v: %v", alias, err)
 				return
 			}
-			if !(tw.tabletFilter == nil || tw.tabletFilter.IsIncluded(tablet.Tablet)) {
-				return
-			}
```

Comment:
I can see why we added the other two locations. However, It's not clear to me how short circuiting this anonymous goroutine/function isn't also helpful? It still avoids unnecessarily getting the mutex and modifying the newTablets structure while holding it, doesn't it?

Reply:
we can't avoid the first run of this goroutine because all we have during the first run is tabletAliases, which don't contain the keyspace or shard information that we need to apply filtering. On subsequent runs, the filtered out tablets will be in the topology watcher's cache as a known tablet and having `refreshKnownTablets` set to false would short circuit the `GetTablet` goroutine. are you suggesting that we should do this filtering even when `refreshKnownTablets` is set to true?